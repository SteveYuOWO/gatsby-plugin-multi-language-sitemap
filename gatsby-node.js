"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.resolveSitemapAndIndex = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _path = _interopRequireDefault(require("path"));

var _optionsValidation = require("./options-validation");

var _fs = _interopRequireDefault(require("fs"));

var _internals = require("./internals");

var _sitemapxml = require("./sitemapxml");

var _excluded = ["url"];

function _createForOfIteratorHelperLoose(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (it) return (it = it.call(o)).next.bind(it); if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; return function () { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

exports.pluginOptionsSchema = _optionsValidation.pluginOptionsSchema;

exports.onPostBuild = /*#__PURE__*/function () {
  var _ref3 = (0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee(_ref, _ref2) {
    var graphql, reporter, pathPrefix, output, query, excludes, resolveSiteUrl, resolvePagePath, resolvePages, filterPages, serialize, langs, _yield$graphql, queryRecords, errors, siteUrl, allPages, _pageFilter, filteredPages, messages, serializedPages, _iterator, _step, page, _yield$Promise$resolv, url, rest, sitemapWritePath, sitemapPublicPath;

    return _regenerator.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            graphql = _ref.graphql, reporter = _ref.reporter, pathPrefix = _ref.pathPrefix;
            output = _ref2.output, query = _ref2.query, excludes = _ref2.excludes, resolveSiteUrl = _ref2.resolveSiteUrl, resolvePagePath = _ref2.resolvePagePath, resolvePages = _ref2.resolvePages, filterPages = _ref2.filterPages, serialize = _ref2.serialize, langs = _ref2.langs;
            _context.next = 4;
            return graphql(query);

          case 4:
            _yield$graphql = _context.sent;
            queryRecords = _yield$graphql.data;
            errors = _yield$graphql.errors;
            _context.next = 9;
            return Promise.resolve(resolveSiteUrl(queryRecords)).catch(function (err) {
              return reporter.panic(_internals.REPORTER_PREFIX + " Error resolving Site URL", err);
            });

          case 9:
            siteUrl = _context.sent;

            if (errors) {
              reporter.panic("Error executing the GraphQL query inside gatsby-plugin-sitemap:\n", errors);
            }

            _context.next = 13;
            return Promise.resolve(resolvePages(queryRecords)).catch(function (err) {
              return reporter.panic(_internals.REPORTER_PREFIX + " Error resolving Pages", err);
            });

          case 13:
            allPages = _context.sent;

            if (!Array.isArray(allPages)) {
              reporter.panic(_internals.REPORTER_PREFIX + " The `resolvePages` function did not return an array.");
            }

            reporter.verbose(_internals.REPORTER_PREFIX + " Filtering " + allPages.length + " pages based on " + excludes.length + " excludes");
            _pageFilter = (0, _internals.pageFilter)({
              allPages: allPages,
              filterPages: filterPages,
              excludes: excludes
            }, {
              reporter: reporter
            }), filteredPages = _pageFilter.filteredPages, messages = _pageFilter.messages;
            messages.forEach(function (message) {
              return reporter.verbose(message);
            });
            reporter.verbose(_internals.REPORTER_PREFIX + " " + filteredPages.length + " pages remain after filtering");
            serializedPages = [];
            _iterator = _createForOfIteratorHelperLoose(filteredPages);

          case 21:
            if ((_step = _iterator()).done) {
              _context.next = 37;
              break;
            }

            page = _step.value;
            _context.prev = 23;
            _context.next = 26;
            return Promise.resolve(serialize(page, {
              resolvePagePath: resolvePagePath
            }));

          case 26:
            _yield$Promise$resolv = _context.sent;
            url = _yield$Promise$resolv.url;
            rest = (0, _objectWithoutPropertiesLoose2.default)(_yield$Promise$resolv, _excluded);
            serializedPages.push((0, _extends2.default)({
              shorturl: url,
              // for langs classfication
              url: (0, _internals.prefixPath)({
                url: url,
                siteUrl: siteUrl,
                pathPrefix: pathPrefix
              })
            }, rest));
            _context.next = 35;
            break;

          case 32:
            _context.prev = 32;
            _context.t0 = _context["catch"](23);
            reporter.panic(_internals.REPORTER_PREFIX + " Error serializing pages", _context.t0);

          case 35:
            _context.next = 21;
            break;

          case 37:
            sitemapWritePath = _path.default.join("public", output);
            sitemapPublicPath = _path.default.posix.join(pathPrefix, output);
            return _context.abrupt("return", resolveSitemapAndIndex({
              hostname: siteUrl,
              publicBasePath: sitemapPublicPath,
              destinationDir: sitemapWritePath,
              sourceData: serializedPages,
              langs: langs
            }));

          case 40:
          case "end":
            return _context.stop();
        }
      }
    }, _callee, null, [[23, 32]]);
  }));

  return function (_x, _x2) {
    return _ref3.apply(this, arguments);
  };
}(); // resolve sitemap and index


var resolveSitemapAndIndex = function resolveSitemapAndIndex(_ref4) {
  var hostname = _ref4.hostname,
      _ref4$publicBasePath = _ref4.publicBasePath,
      publicBasePath = _ref4$publicBasePath === void 0 ? './' : _ref4$publicBasePath,
      destinationDir = _ref4.destinationDir,
      sourceData = _ref4.sourceData,
      langs = _ref4.langs;

  // mkdir if not exist
  _fs.default.mkdirSync(destinationDir, {
    recursive: true
  }); // normalize path


  if (!publicBasePath.endsWith('/')) {
    publicBasePath += '/';
  }

  langs = langs.includes('x-default') ? langs : langs.push('x-default') && langs; // pipe items file

  var urlsMap = generateUrlsMap(langs, sourceData);
  var filesInfoArray = generatefilesInfoArray(urlsMap, langs);
  langs = []; // clear langs to filter langs that had no items.

  for (var _iterator2 = _createForOfIteratorHelperLoose(filesInfoArray), _step2; !(_step2 = _iterator2()).done;) {
    var _step2$value = _step2.value,
        lang = _step2$value.lang,
        pageContent = _step2$value.pageContent;

    _fs.default.writeFileSync(_path.default.resolve(destinationDir, lang + '-sitemap.xml'), pageContent);

    langs.push(lang);
  } // pipe index file


  var sitemapIndexLocs = langs.map(function (lang) {
    return hostname + _path.default.normalize(publicBasePath + lang + '-sitemap.xml');
  });
  var sitemapIndexXML = (0, _sitemapxml.generateSitemapIndexXML)(sitemapIndexLocs);

  var sitemapIndexWritePath = _path.default.resolve(destinationDir, "sitemap-index.xml");

  _fs.default.writeFileSync(sitemapIndexWritePath, sitemapIndexXML);

  return {
    sitemapIndexXML: sitemapIndexXML,
    filesInfoArray: filesInfoArray
  };
}; // generate all files info array 
// the data sturcture like this, [{lang:string, fileContent:string}, ]


exports.resolveSitemapAndIndex = resolveSitemapAndIndex;

function generatefilesInfoArray(urlsMap, langs) {
  var pagesContent = [];
  var allData = [];

  for (var _iterator3 = _createForOfIteratorHelperLoose(urlsMap), _step3; !(_step3 = _iterator3()).done;) {
    var _step3$value = _step3.value,
        _ = _step3$value[0],
        source = _step3$value[1];
    var dataCombine = generateXMLBySource(source);
    allData.push(dataCombine);
  }

  for (var _iterator4 = _createForOfIteratorHelperLoose(langs), _step4; !(_step4 = _iterator4()).done;) {
    var lang = _step4.value;
    var pageContentArray = []; // one page's content

    for (var _iterator5 = _createForOfIteratorHelperLoose(allData), _step5; !(_step5 = _iterator5()).done;) {
      var _step5$value = _step5.value,
          xmlArrayDataString = _step5$value[0],
          xmlMapData = _step5$value[1];
      var xmlLoc = void 0;

      if (xmlMapData.get(lang)) {
        xmlLoc = (0, _sitemapxml.wrapWithLoc)(xmlMapData.get(lang));
      } else {
        continue; // not contains this lang
      }

      var urlData = (0, _sitemapxml.wrapWithUrl)(xmlLoc + xmlArrayDataString);
      pageContentArray.push(urlData);
    }

    if (pageContentArray.length === 0) continue;
    var pageContent = (0, _sitemapxml.wrapWithXMLHeader)((0, _sitemapxml.wrapWithUrlset)(pageContentArray.join('')));
    pagesContent.push({
      lang: lang,
      pageContent: pageContent
    });
  }

  return pagesContent;
} // generate all xml array data by source data


function generateXMLBySource(source) {
  var xmlArrayData = [];
  var xmlMapData = new Map();

  for (var _iterator6 = _createForOfIteratorHelperLoose(source), _step6; !(_step6 = _iterator6()).done;) {
    var _step6$value = _step6.value,
        lang = _step6$value.lang,
        url = _step6$value.url;
    var xmlData = (0, _sitemapxml.wrapWithXMLLink)(lang, url);
    xmlArrayData.push(xmlData);
    xmlMapData.set(lang, url);
  }

  return [xmlArrayData.join(''), xmlMapData];
} // for classfication
// the map (k,v) represents (url, langs array). for example, ('/', [en, fr])


function generateUrlsMap(langs, sourceData) {
  // record langs in a map
  var langsMap = new Map();

  for (var _iterator7 = _createForOfIteratorHelperLoose(langs), _step7; !(_step7 = _iterator7()).done;) {
    var lang = _step7.value;
    langsMap.set(lang, true);
  }

  var urlsMap = new Map();

  for (var _iterator8 = _createForOfIteratorHelperLoose(sourceData), _step8; !(_step8 = _iterator8()).done;) {
    var data = _step8.value;
    // classify by short url
    var shorturl = data.shorturl;
    var _lang = shorturl.split('/')[1];
    var hasLang = langsMap.get(_lang);

    if (hasLang) {
      // remove lang prefix
      shorturl = '/' + shorturl.split('/').slice(2).join('/');
    } else {
      _lang = 'x-default';
    }

    if (!urlsMap.get(shorturl)) urlsMap.set(shorturl, []);
    urlsMap.get(shorturl).push((0, _extends2.default)({
      lang: _lang
    }, data));
  }

  return urlsMap;
}