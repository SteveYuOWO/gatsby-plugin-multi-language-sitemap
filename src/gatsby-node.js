import path from "path"
import { pluginOptionsSchema } from "./options-validation"
import fs from "fs"
import { prefixPath, pageFilter, REPORTER_PREFIX } from "./internals"
import { generateSitemapIndexXML, wrapWithLoc, wrapWithUrl, wrapWithUrlset, wrapWithXMLHeader, wrapWithXMLLink } from "./sitemapxml"

exports.pluginOptionsSchema = pluginOptionsSchema

exports.onPostBuild = async (
  { graphql, reporter, pathPrefix },
  {
    output,
    query,
    excludes,
    resolveSiteUrl,
    resolvePagePath,
    resolvePages,
    filterPages,
    serialize,
    langs
  }
) => {
  const { data: queryRecords, errors } = await graphql(query)

  // resolvePages and resolveSiteUrl are allowed to be sync or async. The Promise.resolve handles each possibility
  const siteUrl = await Promise.resolve(
    resolveSiteUrl(queryRecords)
  ).catch(err =>
    reporter.panic(`${REPORTER_PREFIX} Error resolving Site URL`, err)
  )

  if (errors) {
    reporter.panic(
      `Error executing the GraphQL query inside gatsby-plugin-sitemap:\n`,
      errors
    )
  }

  const allPages = await Promise.resolve(
    resolvePages(queryRecords)
  ).catch(err =>
    reporter.panic(`${REPORTER_PREFIX} Error resolving Pages`, err)
  )

  if (!Array.isArray(allPages)) {
    reporter.panic(
      `${REPORTER_PREFIX} The \`resolvePages\` function did not return an array.`
    )
  }

  reporter.verbose(
    `${REPORTER_PREFIX} Filtering ${allPages.length} pages based on ${excludes.length} excludes`
  )

  const { filteredPages, messages } = pageFilter(
    {
      allPages,
      filterPages,
      excludes,
    },
    { reporter }
  )

  messages.forEach(message => reporter.verbose(message))

  reporter.verbose(
    `${REPORTER_PREFIX} ${filteredPages.length} pages remain after filtering`
  )

  const serializedPages = []

  for (const page of filteredPages) {
    try {
      const { url, ...rest } = await Promise.resolve(
        serialize(page, { resolvePagePath })
      )
      serializedPages.push({
        shorturl: url, // for langs classfication
        url: prefixPath({ url, siteUrl, pathPrefix }),
        ...rest,
      })
    } catch (err) {
      reporter.panic(`${REPORTER_PREFIX} Error serializing pages`, err)
    }
  }

  const sitemapWritePath = path.join(`public`, output)
  const sitemapPublicPath = path.posix.join(pathPrefix, output)

  return resolveSitemapAndIndex({
    hostname: siteUrl,
    publicBasePath: sitemapPublicPath,
    destinationDir: sitemapWritePath,
    sourceData: serializedPages,
    langs: langs
  })
}

// resolve sitemap and index
export const resolveSitemapAndIndex = ({ hostname, publicBasePath = './', destinationDir, sourceData, langs }) => {
  // mkdir if not exist
  fs.mkdirSync(destinationDir, { recursive: true });
  // normalize path
  if (!publicBasePath.endsWith('/')) {
    publicBasePath += '/';
  }
  langs = langs.includes('x-default') ? langs: langs.push('x-default') && langs
  // pipe index file
  const sitemapIndexLocs = langs.map(lang => hostname + path.normalize(publicBasePath + lang + '-sitemap.xml'))
  const sitemapIndexXML = generateSitemapIndexXML(sitemapIndexLocs)
  const sitemapIndexWritePath = path.resolve(destinationDir, `sitemap-index.xml`)
  fs.writeFileSync(sitemapIndexWritePath, sitemapIndexXML)
  const urlsMap = generateUrlsMap(langs, sourceData);
  const filesInfoArray= generatefilesInfoArray(urlsMap, langs);
  for(const {fileName, pageContent} of filesInfoArray) {
    fs.writeFileSync(path.resolve(destinationDir, fileName), pageContent);
  }
  return {
    sitemapIndexXML,
    filesInfoArray
  }
}

// generate all files info array 
// the data sturcture like this, [{fileName:string, fileContent:string}, ]
function generatefilesInfoArray(urlsMap, langs) {
  const pagesContent = []
  const allData = []
  for(const [_, source] of urlsMap) {
    const dataCombine = generateXMLBySource(source)
    allData.push(dataCombine)
  }
  for(const lang of langs) {
    const pageContentArray = [] // one page's content
    for(const [xmlArrayDataString, xmlMapData] of allData) {
      let xmlLoc;
      if(xmlMapData.get(lang)) {
        xmlLoc = wrapWithLoc(xmlMapData.get(lang))
      } else {
        continue; // not contains this lang
      }
      const urlData = wrapWithUrl(xmlLoc + xmlArrayDataString)
      pageContentArray.push(urlData)
    }
    if(pageContentArray.length === 0) continue;
    const pageContent = wrapWithXMLHeader(
      wrapWithUrlset(
        pageContentArray.join('')
      )
    )
    const fileName = lang + '-sitemap.xml'
    pagesContent.push({fileName, pageContent})
  }
  return pagesContent;
}

// generate all xml array data by source data
function generateXMLBySource(source) {
  const xmlArrayData = []
  const xmlMapData = new Map()
  for(const {lang, url} of source) {
    const xmlData = wrapWithXMLLink(lang, url)
    xmlArrayData.push(xmlData)
    xmlMapData.set(lang, url)
  }
  return [xmlArrayData.join(''), xmlMapData]
}

// for classfication
// the map (k,v) represents (url, langs array). for example, ('/', [en, fr])
function generateUrlsMap(langs, sourceData) {
  // record langs in a map
  const langsMap = new Map();
  for(const lang of langs) langsMap.set(lang, true);
  const urlsMap = new Map();
  for(const data of sourceData) {
    // classify by short url
    let shorturl = data.shorturl;
    const lang = shorturl.split('/')[1]
    const hasLang = langsMap.get(lang);
    if(hasLang) {
      // remove lang prefix
      shorturl = '/' + shorturl.split('/').slice(2).join('/')
    } else {
      lang = 'x-default'
    }
    if(!urlsMap.get(shorturl)) urlsMap.set(shorturl, []);
    urlsMap.get(shorturl).push({lang, ...data})
  }
  return urlsMap;
}

