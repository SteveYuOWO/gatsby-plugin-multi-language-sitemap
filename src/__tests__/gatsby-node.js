import { onPostBuild } from "../gatsby-node"
import { Joi } from "gatsby-plugin-utils"
import { pluginOptionsSchema } from "../options-validation"

jest.mock(`sitemap`, () => {
  return {
    SitemapIndexStream: function () {
      this.pipe = jest.fn()
      this.write = jest.fn()
      this.end = jest.fn()
    },
    SitemapItemStream: function () {
      this.pipe = jest.fn()
      this.write = jest.fn()
      this.end = jest.fn()
    }
  }
})

jest.mock(`fs`, () => {
  return {
    mkdirSync: jest.fn(),
    createWriteStream: jest.fn()
  }
})

const schema = pluginOptionsSchema({ Joi })

const pathPrefix = ``

const reporter = {
  verbose: jest.fn(),
  panic: jest.fn(),
}

beforeEach(() => {
  global.__PATH_PREFIX__ = ``
})

describe(`gatsby-plugin-sitemap Node API`, () => {
  it(`should succeed with default options`, async () => {
    const graphql = jest.fn()
    graphql.mockResolvedValue({
      data: {
        site: {
          siteMetadata: {
            siteUrl: `http://dummy.url`,
          },
        },
        allSitePage: {
          nodes: [
            {
              path: `/page-1`,
            },
            {
              path: `/page-2`,
            },
          ],
        },
      },
    })
    const result = await onPostBuild(
      { graphql, pathPrefix, reporter },
      await schema.validateAsync({})
    )
    expect(result.size).toBe(1)
  })

  it(`should accept a custom query`, async () => {
    const graphql = jest.fn()
    const siteUrl = `http://dummy.url`
    graphql.mockResolvedValue({
      data: {
        allSitePage: {
          edges: [
            {
              node: {
                path: `/page-1`,
              },
            },
            {
              node: {
                path: `/post/exclude-page`,
              },
            },
          ],
        },
      },
    })
    const customQuery = `
      {
        allSitePage {
          edges {
            node {
              path
            }
          }
        }
    }`
    const options = {
      output: `custom-folder`,
      resolveSiteUrl: () => siteUrl,
      resolvePages: data => data.allSitePage.edges.map(edge => edge.node),
      serialize: (page, { resolvePagePath }) => {
        return {
          url: resolvePagePath(page),
          changefreq: `weekly`,
          priority: 0.8,
        }
      },
      excludes: [`/post/exclude-page`],
      query: customQuery,
    }

    const result = await onPostBuild(
      { graphql, pathPrefix, reporter },
      await schema.validateAsync(options)
    )
    expect(result.size).toBe(1)
  })

  it(`should include path prefix when creating creating index sitemap`, async () => {
    const graphql = jest.fn()
    graphql.mockResolvedValue({
      data: {
        site: {
          siteMetadata: {
            siteUrl: `http://dummy.url`,
          },
        },
        allSitePage: {
          nodes: [
            {
              path: `/page-1`,
            },
            {
              path: `/page-2`,
            },
          ],
        },
      },
    })

    const options = {
      entryLimit: 1,
    }
    const prefix = `/test`
    const result = await onPostBuild(
      { graphql, pathPrefix: prefix, reporter },
      await schema.validateAsync(options)
    )
    expect(result.size).toBe(1)
  })

  it(`should output modified paths to sitemap`, async () => {
    const graphql = jest.fn()
    graphql.mockResolvedValue({
      data: {
        site: {
          siteMetadata: {
            siteUrl: `http://dummy.url`,
          },
        },
        allSitePage: {
          nodes: [
            {
              path: `/page-1`,
            },
            {
              path: `/page-2`,
            },
          ],
        },
      },
    })
    const prefix = `/test`
    const subdir = `/subdir`
    const options = {
      output: subdir,
    }
    const result = await onPostBuild(
      { graphql, pathPrefix: prefix, reporter },
      await schema.validateAsync(options)
    )
    expect(result.size).toBe(1)
  })

  it(`aftership moke data`, async () => {
    const graphql = jest.fn()
    graphql.mockResolvedValue({
      data: {
        site: {
          siteMetadata: {
            siteUrl: `https://www.aftership.com`,
          },
        },
        allSitePage: {
          nodes: [
            {
              "path": "/"
            },
            {
              "path": "/editor"
            },
            {
              "path": "/amp/editor"
            },
            {
              "path": "/amp/apps/mazet"
            },
            {
              "path": "/apps/legion-express"
            },
            {
              "path": "/amp/apps/legion-express"
            },
            {
              "path": "/dev-404-page/"
            },
            {
              "path": "/zh-Hans/features"
            },
            {
              "path": "/zh-Hans/"
            },
            {
              "path": "/fr/features"
            },
            {
              "path": "/fr/"
            }
          ],
        },
      },
    })
    const prefix = `/test`
    const subdir = `/subdir`
    const options = {
      output: subdir,
      entryLimit: 50000,
      langs: ['fr', 'zh-Hans']
    }
    const result = await onPostBuild(
      { graphql, pathPrefix: prefix, reporter },
      await schema.validateAsync(options)
    )
    expect(result.size).toBe(3)
  })
})
