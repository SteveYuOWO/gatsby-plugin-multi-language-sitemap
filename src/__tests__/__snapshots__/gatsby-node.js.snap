// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-sitemap Node API aftership moke data 1`] = `
Object {
  "filesInfoArray": Array [
    Object {
      "fileName": "fr-sitemap.xml",
      "pageContent": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\"><url><loc>https://www.aftership.com/test/fr/</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"zh-Hans\\" href=\\"https://www.aftership.com/test/zh-Hans/\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"fr\\" href=\\"https://www.aftership.com/test/fr/\\" /></url><url><loc>https://www.aftership.com/test/fr/features</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"zh-Hans\\" href=\\"https://www.aftership.com/test/zh-Hans/features\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"fr\\" href=\\"https://www.aftership.com/test/fr/features\\" /></url></urlset>",
    },
    Object {
      "fileName": "zh-Hans-sitemap.xml",
      "pageContent": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\"><url><loc>https://www.aftership.com/test/zh-Hans/</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"zh-Hans\\" href=\\"https://www.aftership.com/test/zh-Hans/\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"fr\\" href=\\"https://www.aftership.com/test/fr/\\" /></url><url><loc>https://www.aftership.com/test/zh-Hans/features</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"zh-Hans\\" href=\\"https://www.aftership.com/test/zh-Hans/features\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"fr\\" href=\\"https://www.aftership.com/test/fr/features\\" /></url></urlset>",
    },
    Object {
      "fileName": "x-default-sitemap.xml",
      "pageContent": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:xhtml=\\"http://www.w3.org/1999/xhtml\\"><url><loc>https://www.aftership.com/test/</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"zh-Hans\\" href=\\"https://www.aftership.com/test/zh-Hans/\\" /><xhtml:link rel=\\"alternate\\" hreflang=\\"fr\\" href=\\"https://www.aftership.com/test/fr/\\" /></url><url><loc>https://www.aftership.com/test/editor</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/editor\\" /></url><url><loc>https://www.aftership.com/test/amp/editor</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/amp/editor\\" /></url><url><loc>https://www.aftership.com/test/amp/apps/mazet</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/amp/apps/mazet\\" /></url><url><loc>https://www.aftership.com/test/apps/legion-express</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/apps/legion-express\\" /></url><url><loc>https://www.aftership.com/test/amp/apps/legion-express</loc><xhtml:link rel=\\"alternate\\" hreflang=\\"x-default\\" href=\\"https://www.aftership.com/test/amp/apps/legion-express\\" /></url></urlset>",
    },
  ],
  "sitemapIndexXML": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><sitemapindex xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\"><sitemap><loc>https://www.aftership.com/test/subdir/fr-sitemap.xml</loc></sitemap><sitemap><loc>https://www.aftership.com/test/subdir/zh-Hans-sitemap.xml</loc></sitemap><sitemap><loc>https://www.aftership.com/test/subdir/x-default-sitemap.xml</loc></sitemap></sitemapindex>",
}
`;
